common-core-open/src/test/java/edu/isi/nlp/files/ZipFilesTest.java

```diff
@@ -64,6 +70,22 @@ public Symbol apply(final String input) {
     assertEquals(ImmutableSet.of(Symbol.from("2016-09-20-01")), source.keySet());
   }
 
+  @Test
+  public void testVistaUtilsKeyValueSource() throws IOException {
+    final ImmutableKeyValueSource<Symbol, ByteSource> source =
+        KeyValueSources.fromVistaUtilsZipKeyValueSource(vistaUtilsZipFile);
+
+    assertEquals(ImmutableSet.of(Symbol.from("A"), Symbol.from("B")), source.keySet());
+
+    ByteSource entryA = source.getRequired(Symbol.from("A"));
+    String valueA = entryA.asCharSource(StandardCharsets.US_ASCII).read();
+    assertEquals(valueA, "this is value A");
+
+    ByteSource entryB = source.getRequired(Symbol.from("B"));
+    String valueB = entryB.asCharSource(StandardCharsets.US_ASCII).read();
+    assertEquals(valueB, "this is value B");
```

Comment:
This felt like a kind of roundabout way of getting the string values, but I didn't see a more obviously straightforward way. Let me know if this is ok

Reply:
This is the right way.  It has to be a little roundabout because the zip key-value source is more generic than just character data.